core/src/main/java/org/apache/druid/indexer/partitions/PartitionsSpec.java

```diff
@@ -41,6 +41,9 @@
   String MAX_ROWS_PER_SEGMENT = "maxRowsPerSegment";
   int HISTORICAL_NULL = -1;
 
+  @JsonIgnore
+  SecondaryPartitionType getType();
```

Comment:
This was originally to easily get some characteristics of the secondary partitioning type but turns out they are not that useful. But, I still find it useful especially when I need to do something based on the secondary partition type. I can use `switch` instead of multiple `if` statements. But maybe this will be unnecessary once I address https://github.com/apache/druid/pull/9274#discussion_r372148217 in the follow-up PR. I'll remove this method or add javadoc later.